presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -48,6 +49,9 @@ public ZOrder(List<Integer> encodingBits)
 
         totalBitLength = encodingBits.stream().mapToInt(Integer::intValue).sum();
         maxBitLength = encodingBits.stream().mapToInt(Integer::intValue).max().getAsInt();
+
+        curveSearcher = ZOrderCurveSearcher.createZOrderCurveSearcher(this, this.encodingBits, totalBitLength, maxBitLength);
+        curveSearcher.initializeFields();
```

Comment:
Please note that ZOrder and ZOrderCurveSearcher has a circular dependency now, which means these two classes are closely coupled. Does ZOrderCurveSearcher really need to refer a ZOrder object?

Reply:
Currently, yes; ZOrder's `decode` method is non-static, so the instance needs to be passed into ZOrderCurveSearcher for it to call the function. The two classes are definitely very close; if it doesn't interfere with clarity/readability, we might even be able to combine the classes again.